src/components/probe/probing.ts

```diff
@@ -41,20 +41,19 @@ export async function probing(
     // Compile headers using handlebars to render URLs that uses previous responses data.
     // In some case such as value is not string, it will be returned as is without being compiled.
     // If the request does not have any headers, then it should skip this process.
-    let { headers } = requestConfig
-    if (headers) {
-      for await (const header of Object.keys(headers)) {
+    if (requestConfig.headers) {
+      for await (const header of Object.keys(requestConfig.headers)) {
         try {
-          const rawHeader = headers[header]
+          const rawHeader = requestConfig.headers[header]
           const renderHeader = Handlebars.compile(rawHeader)
           const renderedHeader = renderHeader({ responses })
 
-          headers = {
-            ...headers,
+          requestConfig.headers = {
+            ...requestConfig.headers,
             [header]: renderedHeader,
           }
         } catch (_) {
-          headers = { ...headers }
+          requestConfig.headers = { ...requestConfig.headers }
```

Comment:
is this line really necessary? i think this does nothing, doesn't it?

Reply:
done